dp-identity-data-migration/groupsdataextraction/group_extraction.go

```diff
@@ -0,0 +1,283 @@
+	GroupName:        "group_name",
+	UserPoolId:       "user_pool_id",
+	Description:      "description",
+	RoleArn:          "role_arn",
+	Precedence:       "precedence",
+	LastModifiedDate: "last_modified_date",
+	CreationDate:     "creation_date",
+}
+
+type user_group_csv struct {
+	Username string
+	Groups   string
+}
+
+var header_user_groups = user_group_csv{
+	Username: "user_name",
+	Groups:   "groups",
+}
+
+func getgroups(zebCli zebedee.Client, s zebedee.Session) (grouplist zebedee.TeamsList, err error) {
+
+	grouplist, err = zebCli.ListTeams(s)
+
+	if err != nil {
+		fmt.Println("get users error!")
+		return grouplist, err
+	}
+
+	return grouplist, nil
+}
+
+func getUsers(zebCli zebedee.Client, s zebedee.Session) (userlist []zebedee.User, err error) {
+
+	userList, err := zebCli.GetUsers(s)
+
+	if err != nil {
+		fmt.Println("get users error!")
+		return nil, err
+	}
+
+	for _, user := range userList {
+		fmt.Println(user)
+	}
+	return userList, nil
+}
+
+func convert_to_slice_group(input group) []string {
+	return []string{
+		input.GroupName,
+		input.UserPoolId,
+		input.Description,
+		input.RoleArn,
+		input.Precedence,
+		input.LastModifiedDate,
+		input.CreationDate,
+	}
+}
+
+func convert_to_slice_group_user(input user_group_csv) []string {
+	return []string{
+		input.Username,
+		input.Groups,
+	}
+}
+
+func readCsvFile(filePath string) int {
+	f, err := os.Open(filePath)
+	if err != nil {
+		fmt.Println("Unable to read input file "+filePath, err)
+	}
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	records, err := csvReader.ReadAll()
+	if err != nil {
+		fmt.Println("Unable to parse file as CSV for "+filePath, err)
+	}
+
+	return len(records)
+}
+
+func getClient(conf config) zebedee.Client {
+	httpCli := zebedee.NewHttpClient(time.Second * 5)
+	return zebedee.NewClient(conf.host, httpCli)
+}
+
+var emptylist []string
+
+func main() {
+
+	conf := readConfig()
+	zebCli := getClient(conf)
+
+	c := zebedee.Credentials{
+		Email:    conf.user,
+		Password: conf.pword,
+	}
+
+	sess, err := zebCli.OpenSession(c)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	groupList, err := getgroups(zebCli, sess)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	groups_csvfile, err := os.Create(conf.groups_filename)
+	if err != nil {
+		fmt.Printf("failed creating file: %s", err)
+		os.Exit(1)
+	}
+	csvwriter := csv.NewWriter(groups_csvfile)
+	csvwriter.Write(convert_to_slice_group(group_header))
+	for _, zebedeegroup := range groupList.Teams {
+		tmp := group{
+			GroupName:        zebedeegroup.Name,
+			UserPoolId:       "",
+			Description:      zebedeegroup.Name,
+			RoleArn:          "",
+			Precedence:       "10",
+			LastModifiedDate: "",
+			CreationDate:     "",
+		}
+		csvwriter.Write(convert_to_slice_group(tmp))
+
+	}
+	csvwriter.Flush()
+	groups_csvfile.Close()
+	fmt.Println("========= ", conf.groups_filename, "file validiation =============")
+	actualrowcount := readCsvFile(conf.groups_filename) - 1
```

Comment:
You already have a open file handle on `groups_csvfile` - why don't you just use this to read the lines written rather that closing it and passing to `readCsvFile(conf.groups_filename)`?

Reply:
The purpose was to check the number of row written to the file; it function is called twice...  for the group and for usergroup. I was trying not to duplicate code